src/Board.java

```diff
@@ -0,0 +1,171 @@
+import java.util.ArrayList;
+
+public class Board implements Comparable<Board>{
+    private int base;
+    private char[][] boardState;
+    public Board(int base, int startRow, int startCol){
+        this.base = base;
+        this.boardState = new char[base][base];
+        for (int row = 0; row < base; row++){
+            for (int col = 0; col < base; col++){
+                if (row == startRow && col == startCol){
+                    boardState[row][col] = '.';
```

Comment:
These three characters should be constants - define like this
private static final char EMPTY_CELL = '.'